mock_provider_test.go

```diff
@@ -1,260 +0,0 @@
-// Code generated by MockGen. DO NOT EDIT.
```

Comment:
Did not see this being used anywhere. Plus we have another `provider_mock_test` which seems to be achieving the same thing. Opted to delete this and update the latter with new MSGraph SDK mock tests